src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,46 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+  }
+
+  bite(animal) {
+    if (animal instanceof Herbivore) {
+      if (!animal.hidden) {
+        animal.health -= 50;
+
+        if (animal.health <= 0) {
+          const died = Animal.alive.indexOf(animal);
+
+          Animal.alive.splice(died, 1);
```

Comment:
The current implementation uses `indexOf` followed by `splice` inside a loop, which is not optimal for performance. Consider using `filter` to create a new array excluding the dead animal, which can then replace the original `Animal.alive` array.

Reply:
pls apply this suggestion using filter method

And let's move filter logic to Animal static method. For instance - `Animal.removeDead` and use it here